src/activity_handlers.cpp

```diff
@@ -292,6 +293,8 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
             g->m.spawn_item( p->pos(), "bone", bones, 0, age );
             p->add_msg_if_player( m_good, _( "You harvest some usable bones!" ) );
         }
+    } else if( meat == "bone_tainted" ) {
```

Comment:
There are two special cases of "bone_tainted" being handled.
It would be a better way to check `meat == "bone_tainted"` somewhere above, unset `meat` in such a case and give skeletons `MF_BONES` and `MF_POISON` flags.